pymatviz/ptable.py

```diff
@@ -700,291 +763,305 @@ def histogram(
+                text_color (str): Value text color.
+                kwargs (dict): Additional keyword arguments to pass to the `ax.text`.
+            """
+            # Update symbol kwargs
+            kwargs = kwargs or {}
+            kwargs.setdefault("fontsize", 12)
+
+            # Add value for each element
+            for element in Element:
+                # Hide f-block
+                if self.hide_f_block and (element.is_lanthanoid or element.is_actinoid):
+                    continue
+
+                # Get axis index by element symbol
+                symbol: str = element.symbol
+                if symbol not in self.data.index:
+                    continue
+
+                row, column = df_ptable.loc[symbol, ["row", "column"]]
+                ax: plt.Axes = self.axes[row - 1][column - 1]
+
+                # Get and format value
+                content = self.data.loc[symbol, Key.heat_val][0]
+                content = f"{content:{text_fmt}}"
+
+                # Pick value text color
+                if text_color == ElemColorMode.element_types:
+                    color = self.get_elem_type_color(symbol, default="black")
+                elif text_color == "AUTO":
+                    color = pick_bw_for_contrast(self.tile_colors[symbol])
+                else:
+                    color = text_color
+
+                ax.text(
+                    *pos,
+                    content,
+                    color=color,
+                    ha="center",
+                    va="center",
+                    transform=ax.transAxes,
+                    **kwargs,
+                )
 
-    for symbol, row, column, *_ in df_ptable.itertuples():
-        if hide_f_block and (row in (6, 7)):
-            continue
+    # Initialize periodic table plotter
+    projector = HMapPTableProjector(
+        data=data,
+        values_show_mode=values_show_mode,
+        log=log,
+        colormap=colormap,
+        plot_kwargs=plot_kwargs,
+        hide_f_block=hide_f_block,
+    )
 
-        period = n_rows - row  # invert row count to make periodic table right side up
-        tile_value = values.get(symbol)
+    # Call child plotter: heatmap
+    child_kwargs = {
+        "cmap": projector.cmap,
+        "norm": projector.norm,
+    }
 
-        # inf (float/0) or NaN (0/0) are expected when passing in values from
-        # ptable_heatmap_ratio
-        if symbol in exclude_elements:
-            color = "white"
-            label = "excl."
-        elif tile_value == np.inf:
-            color = infty_color  # not in denominator
-            label = r"$\infty$"
-        elif pd.isna(tile_value):
-            color = na_color  # neither numerator nor denominator
-            label = r"$0\,/\,0$"
-        elif tile_value == 0:
-            color = zero_color
-            label = str(zero_symbol)
-        else:
-            color = color_map(norm(tile_value))
+    projector.add_child_plots(
```

Comment:
I'm thinking about replacing this calling child plots architecture right now, because it makes adding custom child plotters really hard and anti-intuitive. 

Instead of writing a child plotter, pass it to the plot caller, we might just inherit from the `PTableProjector` and overwrite the `add_child_plots` method.